ftl-fundamentals/calculator_test.go

```
@@ -0,0 +1,237 @@
+			output: 1,
+		},
+		"4 minus 6 equals negative 2": {
+			a:      4,
+			b:      6,
+			output: -2,
+		},
+	}
+
+	for k, v := range tests {
+		got := calc.Subtract(v.a, v.b)
+		if got != v.output {
+			t.Errorf("%s, got %f, want %f", k, got, v.output)
+		}
+	}
+
+	t.Run("Variadic", func(t *testing.T) {
+		var want float64 = -4
+		if got := calc.Subtract(12, 6, 5, 5); got != want {
+			t.Errorf("got %f, want %f for 12, 6, 5, 5", got, want)
+		}
+	})
+}
+
+func TestMultiply(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		a, b, output float64
+	}{
+		"2 times three equals 6": {
+			a:      2,
+			b:      3,
+			output: 6,
+		},
+		"1st mult arithmetic axiom 0": {
+			a:      5,
+			b:      0,
+			output: 0,
+		},
+		"2nd mult arithmetic axiom -1": {
+			a:      -1,
+			b:      -1,
+			output: 1,
+		},
+	}
+
+	for k, v := range tests {
+		got := calc.Multiply(v.a, v.b)
+		if got != v.output {
+			t.Errorf("%s, got %f, want %f", k, got, v.output)
+		}
+	}
+
+	t.Run("Variadic with 10 inputs", func(t *testing.T) {
+		var want float64 = 11289600
+		if got := calc.Multiply(12, 6, 2, 4, 7, 8, 2, 5, 7, 5); got != want {
+			t.Errorf("got %f, want %f for 12, 6, 2, 4, 7, 8, 5, 7, 5", got, want)
+		}
+	})
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		a, b, output float64
+		errExpected  bool
+	}{
+		"20 divided by 4 equals 5": {
+			a:           20,
+			b:           4,
+			output:      5,
+			errExpected: false,
+		},
+		"Division by zero": {
+			a:           6,
+			b:           0,
+			output:      0,
+			errExpected: true,
+		},
+	}
+
+	for k, v := range tests {
+		got, err := calc.Divide(v.a, v.b)
+		if err != nil && !v.errExpected {
+			t.Fatalf("%s, error status %w, %v", k, err, v.errExpected)
+		}
+		if got != v.output {
+			t.Errorf("%s, got %f, want %f", k, got, v.output)
+		}
+	}
+
+	t.Run("Variadic", func(t *testing.T) {
+		var want float64 = 1
+		if got, _ := calc.Divide(12, 6, 2); got != want {
```

We're ignoring the error here. That's prohibited under the Standing Orders of the Bitfield Institute of Technology!